### PR TITLE
dev-util/jenkins-bin: version bump 2.20

### PR DESCRIPTION
**Pull request by bdouxx**
What's new in 2.20 (2016/08/28)
    Make Cloud.PROVISION permission independent from Jenkins.ADMINISTER. (issue 37616)
    Allow the use of custom JSON signature validator for Update Site metadata signature checks. (issue 36537)
    Do not process null CRON specifications in build triggers. (issue 36748, enhances fix in 2.15)
    Setup wizard now checks if the restart is supported on the system before displaying the restart button. (issue 33374)
    Test Windows junctions before Java 7 symlink in symbolic link checks. (issue 29956)
    Fixed background color in the ComboBoxList element in order to make options visible. (issue 37549)
    Fixed editing default view description with automatic refresh. System message is not being displayed instead of the view description. (issue 37360)
    Fixed process tree management logic on Solaris with 64-bit JVMs. (issue 37559)

What's new in 2.19 (2016/08/21)
    Prevent File descriptor leaks when reading plugin manifests. It causes failures during the upgrade of detached plugins on Windows. (issue 37332, regression in 2.16)
    Prevent resource leaks in AntClassLoader being used in the core. (issue 37561)
    Fix the wrong message about empty field in the case duplicate item name in the New Item dialog. (issue 34532)
    Allow invoking Upgrade Wizard when Jenkins starts up. It can be done by placing an empty jenkins.install.InstallUtil.lastExecVersion file in JENKINS_HOME. (issue 37438)
    Replace repetitious "website" and "dependencies" text in the Setup Wizard by icons. (issue 37523)
    Expose Job name to system logs when Jenkins fails to create a new build with IllegalStateException. (issue 33549)
    Downgrade Queue#maintain() message for dead executors during task mapping from INFO to FINE. (PR #2510)

What's new in 2.18 (2016/08/15)
    Better diagnostics and robustness against old ChangeLogAnnotator API usage in plugins. Enhances JENKINS-23365 fix in 1.569. (issue 36757)
    Prevent open file leak when the agent channel onClose() listener writes to the already closed log. (issue 37098)
    Stop A/B testing of the remoting JNLP3 protocol due to the known issues. The protocol can be enabled manually via the jenkins.slaves.JnlpSlaveAgentProtocol3.enabled system property. (issue 37315)
    When checking Update Center, append ?uctest parameter to HTTP and HTTPS URLs only. (issue 37189)
    Incorrect formatting of messages in the Upate Center and Setup Wizard. (issue 36757)
    Massive cleanup of issues reported by FindBugs. User-visible issues - wrong log message formatting bugs in the Update Center and user creation logic. (issue 36717, PR #2459)
    Remoting 2.61: JNLP Slave connection issue with JNLP3-connect when the generated encrypted cookie contains a newline symbols. (issue 37140)
    Remoting 2.61: Retry loading classes when remote classloader gets interrupted. (issue 36991)
    Remoting 2.61: Improve diagnostics of Local Jar Cache write errors. (remoting PR #91)
    Remoting 2.62: Be robust against the delayed EOF command when unexporting input and output streams. (issue 22853)
    Remoting 2.62: Cleanup of minor issues discovered by FindBugs. (remoting PR #96)
    Remoting 2.62: Improve class filtering performance in remote invocations. (issue 37218)
    Remoting 2.62: TCP agent connection listener now publishes a list of supported agent protocols to speed up the connection setup. (issue 37031)
    Improve German, Lithuanian and Bulgarian translations. (PR #2473, PR #2470, PR #2498 ) 

Package-Manager: portage-2.2.28

### CODE REVIEW

File: dev-util/jenkins-bin/jenkins-bin-2.20.ebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+SRC_URI="http://mirrors.jenkins-ci.org/war/${PV}/${PN/-bin/}.war -> ${P}.war"
+RESTRICT="mirror"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux"
+IUSE=""
+
+RDEPEND="media-fonts/dejavu
+	media-libs/freetype
+	!dev-util/jenkins-bin:lts
+	>=virtual/jre-1.7.0"
```
Comment (monsieurp) [inline dev-util/jenkins-bin/jenkins-bin-2.20.ebuild]: You've got to `inherit java-pkg-2` for this check to work.

**Comment by gktrk** [comment]
@gentoo/proxy-maint @patricklauer

File: dev-util/jenkins-bin/jenkins-bin-2.20.ebuild (hunk 1 above)
Comment (bdouxx) [inline dev-util/jenkins-bin/jenkins-bin-2.20.ebuild]: it's a binary package (just copy war file), is it really necessary?

Comment (monsieurp) [inline dev-util/jenkins-bin/jenkins-bin-2.20.ebuild]: Yes. We need to make sure there's indeed a JVM installed and that's what the `java-pkg-2` eclass is there for.

**Comment by bdouxx** [comment]
Don't commit my second commit. it' s the first time I'm try to correct a commit, and I didn't know that github automatically make the pull request...

I have a problem if I add java-pkg-2 in the ebuild file as it:

inherit user systemd java-pkg-2

 * ERROR: dev-util/jenkins-bin-2.21::bdouxx failed (prepare phase):
 *   eapply_user (or default) must be called in src_prepare()!
 * 
 * Call stack:
 *            ebuild.sh, line  780:  Called __ebuild_main 'prepare'
 *   phase-functions.sh, line 1005:  Called __dyn_prepare
 *   phase-functions.sh, line  380:  Called die
 * The specific snippet of code:
 *              die "eapply_user (or default) must be called in src_prepare()!"


do you have any idea how to solve this?

**Comment by bdouxx** [comment]
hi,
I think it's ok now, work well for me.

**Comment by monsieurp** [comment]
Thank you for the PR but dd49531dfeae0f7d4e7e2bd986df7fc4c9ab107a got there first. Closing.

**Pull request closed without merging**